sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -65,6 +68,18 @@ def _need_new_token(self):
         return not self._token or self._token.expires_on - time.time() < 300
 
 
+    def _need_adding_header(self, url):
```

Comment:
This name doesn't resonate with me. You are trying to guard against the domain changing, right? Maybe something like "domain_changed"?

Reply:
It is more than domain change check. If _always_adding_header is set to True, we skip the domain parsing step and return True directly.

Are you suggesting moving the if (_always_adding_header) out of the method?

Reply:
> Are you suggesting moving the if (_always_adding_header) out of the method?

Yes.